ospfd/ospf_asbr.c

```diff
@@ -275,6 +275,30 @@ void ospf_asbr_status_update(struct ospf *ospf, uint8_t status)
 	ospf_router_lsa_update(ospf);
 }
 
+/* If there's redistribution configured, we need to refresh external
+ * LSAs in order to install Type-7 and flood to all NSSA Areas
+ */
+void ospf_asbr_nssa_redist_task(struct ospf *ospf)
+{
+	int type;
+
+	for (type = 0; type < ZEBRA_ROUTE_MAX; type++) {
+		struct list *red_list;
+		struct listnode *node;
+		struct ospf_redist *red;
+
+		red_list = ospf->redist[type];
+		if (!red_list)
+			continue;
+
+		for (ALL_LIST_ELEMENTS_RO(red_list, node, red))
+			ospf_external_lsa_refresh_type(
+				ospf, type, red->instance, LSA_REFRESH_FORCE);
```

Comment:
I might be missing something here. Consider a case where type5 LSA has already been originated. We should flush the same before originatig type-7. ospf_external_lsa_refresh_type function does not seems to do that.

Reply:
I added flushing before refreshing but it doesn't seem to be working (this PR was updated, you can look at the code). The flush happens when the area is already an NSSA area. MaxAge Type-5 LSA's will not be flooded via ospf_flood_through_as() because of this:
```
int ospf_flood_through_as(struct ospf *ospf, struct ospf_neighbor *inbr,
			  struct ospf_lsa *lsa)
{
...
		switch (area->external_routing) {
		/* Don't send AS externals into stub areas.  Various types
		   of support for partial stub areas can be implemented
		   here.  NSSA's will receive Type-7's that have areas
		   matching the originl LSA. */
		case OSPF_AREA_NSSA: /* Sending Type 5 or 7 into NSSA area */
				     /* Type-7, flood NSSA area */
			if (lsa->data->type == OSPF_AS_NSSA_LSA
			    && area == lsa->area)
				/* We will send it. */
				continue_flag = 0;
			else
				continue_flag = 1; /* Skip this NSSA area for
						      Type-5's et al */
			break;
```

Reply:
I returned back the version without flushing. Any external LSA's received or originated before enabling NSSA will be handled by the MaxAge timer.